src/shader_recompiler/ir/passes/resource_tracking_pass.cpp

```diff
@@ -539,28 +543,49 @@ void PatchImageSampleInstruction(IR::Block& block, IR::Inst& inst, Info& info,
         }
     }();
 
+    const auto unnormalized = sampler.force_unnormalized || inst_info.is_unnormalized;
+    // Query dimensions of image if needed for normalization.
+    // We can't use the image sharp because it could be bound to a different image later.
+    const auto dimensions =
+        unnormalized ? ir.ImageQueryDimension(ir.Imm32(image_binding), ir.Imm32(0u), ir.Imm1(false))
+                     : IR::Value{};
+    const auto fix_coord = [&](IR::F32 coord, u32 dim_idx) -> IR::Value {
```

Comment:
Could probably name this get_coord and also call get_addr_reg inside it, so you only call this in switch statement below

Reply:
Good idea, done.